### PR TITLE
Add a few trait impls for AccessError

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by stjepang** [comment]
There were some submodule updates, but now they seem to be fixed. Sorry for the noise!

File: src/libstd/thread/local.rs [hunk 1]
```diff
@@ -207,6 +209,8 @@ impl fmt::Display for AccessError {
     }
 }
 
+impl Error for AccessError {}
```
Comment (Mark-Simulacrum) [inline src/libstd/thread/local.rs]: We'll probably want a `#[stable...` attribute on this even though it does nothing today :)

**Comment by dtolnay** [comment]
@rust-lang/libs -- this PR adds impls of Clone, Copy, Eq, PartialEq, and Error for [std::thread::AccessError](https://doc.rust-lang.org/std/thread/struct.AccessError.html), which is the error returned by the try_with method on thread locals. See [Utf8Error](https://doc.rust-lang.org/std/str/struct.Utf8Error.html), [CharTryFromError](https://doc.rust-lang.org/std/char/struct.CharTryFromError.html), [TryFromIntError](https://doc.rust-lang.org/std/num/struct.TryFromIntError.html), [fmt::Error](https://doc.rust-lang.org/std/fmt/struct.Error.html), [RecvTimeoutError](https://doc.rust-lang.org/std/sync/mpsc/enum.RecvTimeoutError.html), [RecvError](https://doc.rust-lang.org/std/sync/mpsc/struct.RecvError.html), [TryRecvError](https://doc.rust-lang.org/std/sync/mpsc/enum.TryRecvError.html) as examples of existing error types having all of these impls.

```diff
  impl<T: 'static> LocalKey<T> {
      pub fn try_with<F, R>(&'static self, f: F) -> Result<R, AccessError>
      where
          F: FnOnce(&T) -> R;
  }

  pub struct AccessError { _private: () }

+ impl Clone for AccessError {...}
+ impl Copy for AccessError {...}
+ impl Eq for AccessError {...}
+ impl PartialEq for AccessError {...}
+ impl std::error::Error for AccessError {...}
```

@rfcbot fcp merge

**Pull request merged**